bson/src/main/org/bson/codecs/pojo/PojoBuilderHelper.java

```diff
@@ -149,15 +157,30 @@ private static TypeParameterMap getTypeParameterMap(final List<String> genericTy
     }
 
     @SuppressWarnings("unchecked")
-    private static <T> TypeData<T> getFieldTypeDataFromClass(final Class<T> type) {
```

Comment:
This was used when we weren't reflecting over the class in the same way and the type data was overwritten by looking at the field in the context of the class model.